configure

```diff
@@ -73,17 +73,43 @@ EOF
 # This is not an endorsement of specific expiration limits, key sizes, or algorithms.
 generate_tls_dev_cert() {
     if [ ! -e "${rootdir}/dev/erlserver.pem" ]; then
-        openssl req -newkey rsa:2048 -new -nodes -x509 -days 3650 -keyout key.pem -out cert.pem
-        cat key.pem cert.pem > dev/erlserver.pem && rm key.pem cert.pem
+        openssl genrsa 2048 > ca-key.pem
+        openssl req -new -x509 -nodes -days 3650 -key ca-key.pem -out ca-cert.pem
+        openssl req -newkey rsa:2048 -nodes -days 3650 -keyout key.pem -out req.pem
+        openssl x509 -req -days 3650 -set_serial 01 -in req.pem -out cert.pem -CA ca-cert.pem -CAkey ca-key.pem
+        openssl verify -CAfile ca-cert.pem ca-cert.pem cert.pem
+        cat key.pem cert.pem > dev/erlserver.pem && mv ca-cert.pem dev/ && rm ca-key.pem key.pem cert.pem req.pem
     fi
 
     if [ ! -e "${rootdir}/dev/couch_ssl_dist.conf" ]; then
         cat > "${rootdir}/dev/couch_ssl_dist.conf" << EOF
-[{server,
-  [{certfile, "${rootdir}/dev/erlserver.pem"},
-   {secure_renegotiate, true}]},
- {client,
-  [{secure_renegotiate, true}]}].
+[{server, [
+    {certfile, "${rootdir}/dev/erlserver.pem"},
+    {secure_renegotiate, true}
+  ]},
+  {client, [
+    {secure_renegotiate, true},
+    {verify, verify_peer},
+    {cacertfile, "${rootdir}/dev/ca-cert.pem"},
+    {fail_if_no_peer_cert, true},
+    {verify_fun, {
```

Comment:
Can you also take a look into customize_hostname_check. I don't know when it was introduced. But it seems to be a better way.

Reply:
I tried `{customize_hostname_check, [{match_fun, public_key:pkix_verify_hostname_match_fun(https)}]}` but it didn't work.

Reply:
Did you try making the match_fun always passing just to test that function is available in the OTP we are using.

define and export a function in any module

```
always_true() ->
   fun(_) -> true.
```

Then add configuration `{customize_hostname_check, [{match_fun, your_module:always_true()]}`. If it works, look into how https://github.com/erlang/otp/blob/2c864f2f9ef61750006afde6af33453a11d025cd/lib/public_key/src/public_key.erl#L1296 is defined in recent OTP and compare it with the OTP version we use. Understand the difference. If new OTP version has a fix we can make a conditional based on OTP version used. For older versions we would use our own function (inspired by latest OTP). On newer versions of OTP we would use function provided by OTP.

In any case we shouldn't not disable security by keeping `match_fun, your_module:always_true()` or `{verify, verify_none}`. Our goal is to suppress the warning not to disable security.

Reply:
The problem with this approach is that the TLS distribution starts earlier than other modules, even earlier than the standard io drivers. If we use custom module function in conf file, it will throw error about `undefined function`.